### ISSUE TITLE
adaptive_uniform_ratio is not a uniform floor: effective strength scales with num_envs and failure rate

### ISSUE THREAD

**Issue by linjiw** [HybridRobotics/whole_body_tracking#73]
## Summary

In the tracking task's adaptive sampler, the uniform prior is combined with the failure statistic **additively**, and the failure statistic is an EMA of raw *counts*. The uniform term's share of the sampling distribution is therefore `ε / (Σq + ε)`, which shrinks as failures accumulate and as the number of environments grows — not the fixed `ε` that `adaptive_uniform_ratio` implies.

Practical consequence: `adaptive_uniform_ratio` provides **no lower bound** on any bin's sampling probability, and the same configured value behaves very differently at different `--num_envs`. Since the repo exposes `--adaptive_uniform_ratio` as a tuning flag, anyone who tuned it at one scale and then changed `num_envs` has silently changed their sampler.

## Where

`source/whole_body_tracking/whole_body_tracking/tasks/tracking/mdp/commands.py:217`:

```python
sampling_probabilities = self.bin_failed_count + self.cfg.adaptive_uniform_ratio / float(self.bin_count)
```

with `bin_failed_count` maintained as an EMA of `torch.bincount(fail_bins, ...)`, so it carries units of *episodes failed per update* and scales with the environment count, rather than being a normalised distribution.

## Derivation

Let `q` be the failure EMA over `N` bins and `ε = adaptive_uniform_ratio`:

```
p_i = (q_i + ε/N) / (Σq + ε)
```

Total mass contributed by the uniform term:

```
Σ_i (ε/N) / (Σq + ε)  =  ε / (Σq + ε)
```

which equals `ε` only when `Σq ≈ 1`:

| Σq (mean failures/update) | uniform share at ε = 0.1 |
|---:|---:|
| 0.1 | 50% |
| 1 | 9.1% |
| 10 | 1.0% |
| 50 | 0.2% |

## Minimal reproduction

No simulator required:

```python
import torch

N, eps = 100, 0.1
for scale in (0.1, 1.0, 10.0, 50.0):
    q = torch.zeros(N)
    q[0] = scale                      # failure mass concentrated on one bin
    additive = (q + eps / N) / (q + eps / N).sum()
    mixture  = (1 - eps) * (q / q.sum()) + eps / N
    print(f"Sum(q)={scale:5.1f}  additive: top1={additive.max():.3f}, "
          f"uniform share={eps/(q.sum()+eps):.3%}   "
          f"mixture: top1={mixture.max():.3f}, uniform share={eps:.1%}")
```

```
Sum(q)=  0.1  additive: top1=0.505, uniform share=50.000%   mixture: top1=0.901, uniform share=10.0%
Sum(q)=  1.0  additive: top1=0.910, uniform share=9.091%   mixture: top1=0.901, uniform share=10.0%
Sum(q)= 10.0  additive: top1=0.990, uniform share=0.990%   mixture: top1=0.901, uniform share=10.0%
Sum(q)= 50.0  additive: top1=0.998, uniform share=0.200%   mixture: top1=0.901, uniform share=10.0%
```

The additive form swings from 0.505 to 0.998 top-1 mass across a plausible range of failure rates; the mixture is invariant.

In a clip-level port of this sampler (units = motion clips rather than time bins, 4096 envs, `ε = 0.1`, `adaptive_alpha = 0.01`) the distribution collapsed to a single unit holding 87–89% of all sampling mass in every seed, entropy falling to ~0.12–0.15, despite the nominal "10% uniform floor". A convex mixture held top-1 mass at ≤ 0.19 under otherwise identical settings.

## Suggested fix

Normalise before mixing, so `ε` becomes a scale-invariant coverage floor:

```python
q = self.bin_failed_count
q = q / q.sum() if float(q.sum()) > 0 else torch.full_like(q, 1.0 / self.bin_count)
sampling_probabilities = (1.0 - eps) * q + eps / self.bin_count
```

This is behaviour-changing for existing configs, so a flag defaulting to current behaviour may be preferable. A documentation-only alternative is to state that `adaptive_uniform_ratio` is an additive pseudo-count rather than a floor, whose effective weight depends on `num_envs` and the failure rate.

Happy to open a PR for whichever you prefer.

## Related

The same expression appears in mjlab's tracking task; filed there as mujocolab/mjlab#1153 with the same derivation and repro.
